src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/identifier/UnquoteIdentifierRule.java

```diff
@@ -0,0 +1,91 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.rewriter.identifier;
+
+import com.alibaba.druid.sql.ast.expr.SQLIdentifierExpr;
+import com.alibaba.druid.sql.ast.expr.SQLPropertyExpr;
+import com.alibaba.druid.sql.ast.expr.SQLQueryExpr;
+import com.alibaba.druid.sql.ast.statement.SQLExprTableSource;
+import com.alibaba.druid.sql.ast.statement.SQLSelectItem;
+import com.alibaba.druid.sql.dialect.mysql.visitor.MySqlASTVisitorAdapter;
+import com.amazon.opendistroforelasticsearch.sql.rewriter.RewriteRule;
+import com.amazon.opendistroforelasticsearch.sql.utils.BackticksUnquoter;
+
+public class UnquoteIdentifierRule extends MySqlASTVisitorAdapter implements RewriteRule<SQLQueryExpr> {
+
+    String identifier = null;
+
+    @Override
+    public boolean visit(SQLPropertyExpr propertyExpr) {
+        String identifier = ((SQLIdentifierExpr) propertyExpr.getOwner()).getName();
+        if (identifier.startsWith("`") && identifier.endsWith("`")) {
+            this.identifier = new BackticksUnquoter().unquoteSingleField(identifier) + "."
+                    + new BackticksUnquoter().unquoteSingleField(propertyExpr.getName());
+            SQLSelectItem selectItem = (SQLSelectItem) propertyExpr.getParent();
+            selectItem.setExpr(new SQLIdentifierExpr(this.identifier));
+            this.identifier = null;
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public boolean visit(SQLSelectItem selectItem) {
+        try {
+            String identifier = ((SQLIdentifierExpr) selectItem.getExpr()).getName();
```

Comment:
This seems very strange. Please add proper conditions here, and not just ignore errors by hiding them in finally part

Reply:
@galkk ok will do